File: MiniDOMOffice/Presenze/RichiestePermessiFerie/RichiestaPermessoFerieCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {

        /// <summary>
        /// Repository di <see cref="RichiestaPermessoFerie"/>
        /// </summary>
        [Serializable]
        public class RichiestaPermessoFerieCursor 
            : minidom.Databases.DBObjectCursorPO<RichiestaPermessoFerie>
        {
            private DBCursorField<DateTime> m_DataRichiesta = new DBCursorField<DateTime>("DataRichiesta");
            private DBCursorField<DateTime> m_DataInizio = new DBCursorField<DateTime>("DataInizio");
            private DBCursorField<DateTime> m_DataFine = new DBCursorField<DateTime>("DataFine");
            private DBCursorStringField m_MotivoRichiesta = new DBCursorStringField("MotivoRichiesta");
            private DBCursorStringField m_NoteRichiesta = new DBCursorStringField("NoteRichiesta");
            private DBCursorField<TipoRichiestaPermessoFerie> m_TipoRichiesta = new DBCursorField<TipoRichiestaPermessoFerie>("TipoRichiesta");
            private DBCursorField<int> m_IDRichiedente = new DBCursorField<int>("IDRichiedente");
            private DBCursorStringField m_NomeRichiedente = new DBCursorStringField("NomeRichiedente");
            private DBCursorField<DateTime> m_DataPresaInCarico = new DBCursorField<DateTime>("DataPresaInCarico");
            private DBCursorField<int> m_IDInCaricoA = new DBCursorField<int>("IDInCaricoA");
            private DBCursorStringField m_NomeInCaricoA = new DBCursorStringField("NomeInCaricoA");
            private DBCursorField<EsitoRichiestaPermessoFerie> m_EsitoRichiesta = new DBCursorField<EsitoRichiestaPermessoFerie>("EsitoRichiesta");
            private DBCursorStringField m_DettaglioEsitoRichiesta = new DBCursorStringField("DettaglioEsitoRichiesta");
            private DBCursorStringField m_NotaPrvSupervisore = new DBCursorStringField("NotaPrvSupervisore");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");

            /// <summary>
            /// Costruttore
            /// </summary>
            public RichiestaPermessoFerieCursor()
            {
            }

            /// <summary>
            /// DataRichiesta
            /// </summary>
            public DBCursorField<DateTime> DataRichiesta
            {
                get
                {
                    return m_DataRichiesta;
                }
            }

            /// <summary>
            /// DataInizio
            /// </summary>
            public DBCursorField<DateTime> DataInizio
            {
                get
                {
                    return m_DataInizio;
                }
            }

            /// <summary>
            /// NotaPrvSupervisore
            /// </summary>
            public DBCursorStringField NotaPrvSupervisore
            {
                get
                {
                    return m_NotaPrvSupervisore;
                }
            }

            /// <summary>
            /// DataFine
            /// </summary>
            public DBCursorField<DateTime> DataFine
            {
                get
                {
                    return m_DataFine;
                }
            }

            /// <summary>
            /// MotivoRichiesta
            /// </summary>
            public DBCursorStringField MotivoRichiesta
            {
                get
                {
                    return m_MotivoRichiesta;
                }
            }

            /// <summary>
            /// NoteRichiesta
            /// </summary>
            public DBCursorStringField NoteRichiesta
            {
                get
                {
                    return m_NoteRichiesta;
                }
            }

            /// <summary>
            /// TipoRichiesta
            /// </summary>
            public DBCursorField<TipoRichiestaPermessoFerie> TipoRichiesta
            {
                get
                {
                    return m_TipoRichiesta;
                }
            }

            /// <summary>
            /// IDRichiedente
            /// </summary>
            public DBCursorField<int> IDRichiedente
            {
                get
                {
                    return m_IDRichiedente;
                }
            }

            /// <summary>
            /// NomeRichiedente
            /// </summary>
            public DBCursorStringField NomeRichiedente
            {
                get
                {
                    return m_NomeRichiedente;
                }
            }

            /// <summary>
            /// DataPresaInCarico
            /// </summary>
            public DBCursorField<DateTime> DataPresaInCarico
            {
                get
                {
                    return m_DataPresaInCarico;
                }
            }

            /// <summary>
            /// IDInCaricoA
            /// </summary>
            public DBCursorField<int> IDInCaricoA
            {
                get
                {
                    return m_IDInCaricoA;
                }
            }

            /// <summary>
            /// NomeInCaricoA
            /// </summary>
            public DBCursorStringField NomeInCaricoA
            {
                get
                {
                    return m_NomeInCaricoA;
                }
            }

            /// <summary>
            /// EsitoRichiesta
            /// </summary>
            public DBCursorField<EsitoRichiestaPermessoFerie> EsitoRichiesta
            {
                get
                {
                    return m_EsitoRichiesta;
                }
            }

            /// <summary>
            /// DettaglioEsitoRichiesta
            /// </summary>
            public DBCursorStringField DettaglioEsitoRichiesta
            {
                get
                {
                    return m_DettaglioEsitoRichiesta;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }
 
             
            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.RichiestePermessiFerie;
            }
        }
    }
}

File: MiniDOMOffice/Presenze/RichiestePermessiFerie/RichiestaPermessoFerieCursor.vb
Imports minidom.Anagrafica
Imports minidom.Databases
Imports minidom.Sistema


Partial Class Office


    Public Class RichiestaPermessoFerieCursor
        Inherits DBObjectCursorPO(Of RichiestaPermessoFerie)

        Private m_DataRichiesta As New CCursorField(Of Date)("DataRichiesta")
        Private m_DataInizio As New CCursorField(Of Date)("DataInizio")
        Private m_DataFine As New CCursorField(Of Date)("DataFine")
        Private m_MotivoRichiesta As New CCursorFieldObj(Of String)("MotivoRichiesta")
        Private m_NoteRichiesta As New CCursorFieldObj(Of String)("NoteRichiesta")

        Private m_TipoRichiesta As New CCursorField(Of TipoRichiestaPermessoFerie)("TipoRichiesta")

        Private m_IDRichiedente As New CCursorField(Of Integer)("IDRichiedente")
        Private m_NomeRichiedente As New CCursorFieldObj(Of String)("NomeRichiedente")

        Private m_DataPresaInCarico As New CCursorField(Of Date)("DataPresaInCarico")
        Private m_IDInCaricoA As New CCursorField(Of Integer)("IDInCaricoA")
        Private m_NomeInCaricoA As New CCursorFieldObj(Of String)("NomeInCaricoA")
        Private m_EsitoRichiesta As New CCursorField(Of EsitoRichiestaPermessoFerie)("EsitoRichiesta")
        Private m_DettaglioEsitoRichiesta As New CCursorFieldObj(Of String)("DettaglioEsitoRichiesta")
        Private m_NotaPrvSupervisore As New CCursorFieldObj(Of String)("NotaPrvSupervisore")

        Private m_Flags As New CCursorField(Of Integer)("Flags")

        Public Sub New()
        End Sub

        Public ReadOnly Property DataRichiesta As CCursorField(Of Date)
            Get
                Return Me.m_DataRichiesta
            End Get
        End Property

        Public ReadOnly Property DataInizio As CCursorField(Of Date)
            Get
                Return Me.m_DataInizio
            End Get
        End Property

        Public ReadOnly Property NotaPrvSupervisore As CCursorFieldObj(Of String)
            Get
                Return Me.m_NotaPrvSupervisore
            End Get
        End Property


        Public ReadOnly Property DataFine As CCursorField(Of Date)
            Get
                Return Me.m_DataFine
            End Get
        End Property

        Public ReadOnly Property MotivoRichiesta As CCursorFieldObj(Of String)
            Get
                Return Me.m_MotivoRichiesta
            End Get
        End Property

        Public ReadOnly Property NoteRichiesta As CCursorFieldObj(Of String)
            Get
                Return Me.m_NoteRichiesta
            End Get
        End Property

        Public ReadOnly Property TipoRichiesta As CCursorField(Of TipoRichiestaPermessoFerie)
            Get
                Return Me.m_TipoRichiesta
            End Get
        End Property

        Public ReadOnly Property IDRichiedente As CCursorField(Of Integer)
            Get
                Return Me.m_IDRichiedente
            End Get
        End Property

        Public ReadOnly Property NomeRichiedente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeRichiedente
            End Get
        End Property

        Public ReadOnly Property DataPresaInCarico As CCursorField(Of Date)
            Get
                Return Me.m_DataPresaInCarico
            End Get
        End Property

        Public ReadOnly Property IDInCaricoA As CCursorField(Of Integer)
            Get
                Return Me.m_IDInCaricoA
            End Get
        End Property

        Public ReadOnly Property NomeInCaricoA As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeInCaricoA
            End Get
        End Property

        Public ReadOnly Property EsitoRichiesta As CCursorField(Of EsitoRichiestaPermessoFerie)
            Get
                Return Me.m_EsitoRichiesta
            End Get
        End Property

        Public ReadOnly Property DettaglioEsitoRichiesta As CCursorFieldObj(Of String)
            Get
                Return Me.m_DettaglioEsitoRichiesta
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeRichiestePermF"
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Office.RichiestePermessiFerie.Module
        End Function

    End Class


End Class